tss-esapi/src/context/general_esys_tr.rs

```diff
@@ -388,6 +392,50 @@ impl Context {
         TpmHandle::try_from(tpm_handle)
     }
 
-    // Missing function: Esys_TR_Serialize
-    // Missing function: Esys_TR_Deserialize
+    /// Store the `ObjectHandle` in a buffer
+    pub fn tr_serialize(&mut self, handle: ObjectHandle) -> Result<Vec<u8>> {
+        let mut len = 0;
+        let mut buffer: *mut u8 = null_mut();
+        let mut result = vec![];
+        ReturnCode::ensure_success(
+            unsafe { Esys_TR_Serialize(self.mut_context(), handle.into(), &mut buffer, &mut len) },
+            |ret| {
+                error!("Error while serializing handle: {}", ret);
+            },
+        )?;
+        unsafe {
```

Comment:
This feels like it should belong in a function in ```ffi.rs```.

(some psedu rust code):
```rust
pub(crate) fn to_owned_bytes(ffi_bytes_ptr: *mut u8, size: usize) -> Vec<u8> {
    let ffi_bytes = unsafe {MBox::<[u8]>::from_raw_parts(ffi_bytes_ptr, size) };
    return Vec::<u8>::from(ffi_bytes.as_ref())
}
```

And then the Context method would be something like:
```rust
    /// Store the `ObjectHandle` in a buffer
    pub fn tr_serialize(&mut self, handle: ObjectHandle) -> Result<Vec<u8>> {
        let mut len = 0;
        let mut buffer_ptr: *mut u8 = null_mut();
        ReturnCode::ensure_success(
            unsafe { Esys_TR_Serialize(self.mut_context(), handle.into(), &mut buffer_ptr, &mut len) },
            |ret| {
                error!("Error while serializing handle: {}", ret);
            },
        )?;
        Ok(to_owned_buffer(
                buffer_ptr,
                len.try_into().map_err(|e| {
                    error!("Failed to convert buffer len to usize: {}", e);
                    Error::local_error(WrapperErrorKind::InvalidParam)
                })?,
            )
        )
    }
```